docs/content/direct/multi-wec-aggregated-status.md

```diff
@@ -0,0 +1,171 @@
+   For workloads deployed to multiple WECs, the `.status` field in the WDS remains empty, resulting in non-matching WDS and WEC status.
+
+This leads to limited visibility in ArgoCD, where workloads appear as *OutOfSync* or *Unknown*, even though all WECs are healthy.
+
+**Reference:**  
+- [PR #3297](https://github.com/kubestellar/kubestellar/pull/3297)  
+- [Multi-cluster example in docs](https://docs.kubestellar.io/release-0.28.0/direct/example-scenarios/#scenario-1-multi-cluster-workload-deployment-with-kubectl)
+
+---
+
+## Goal
+
+Implement a **multi-WEC aggregated status reporting mechanism** in KubeStellar to reflect combined readiness in the WDS.  
+This will allow aggregated workload visibility across all WECs and ensure compatibility with ArgoCD’s health checks.
+
+Key objectives:
+- Aggregate per-WEC `WorkStatus` objects.
+- Update the `.status` field in WDS workload objects.
+- Maintain backward compatibility with singleton workloads.
+- Provide fallback to existing singleton mechanism when applicable.
+
+---
+
+## Solution Approach
+
+### 1. Extend BindingPolicy CRD
+
+Add a new field `wantMultiWECReportedState`, similar to `wantSingletonReportedState`.
+
+Logic:
+- If **enabled** and number of WECs == 1 → use existing singleton mechanism.
+- If **enabled** and number of WECs > 1 → invoke new multi-WEC aggregation logic.
+- If **disabled**, retain existing empty status behavior.
+
+```yaml
+spec:
+  clusterSelectors:
+  - matchLabels: {"location": "edge"}
+  downsync:
+  - objectSelectors:
+    - matchLabels: {"app.kubernetes.io/name": "nginx-multi"}
+    wantMultiWECReportedState: true
+```
+
+The flag is defined under each downsync rule, consistent with [example scenario 4](https://docs.kubestellar.io/release-0.28.0/direct/example-scenarios/#scenario-4-singleton-status).
+
+---
+
+### 2. Introduce `handleMultiReportedState()` Function
+
+A new controller function will handle multi-WEC aggregation:
+
+**Responsibilities:**
+1. Collect `.status` from all target WECs using existing status collection logic (see [pkg/status/singletonstatus.go#L64](https://github.com/kubestellar/kubestellar/blob/main/pkg/status/singletonstatus.go#L64)).  
+2. Aggregate status data according to defined field rules (numeric, conditions, timestamps, string) or use a per-kind aggregator if it is a known KIND.  
+3. Update the aggregated `.status` back into the WDS workload object.
+
+**Focus:** Aggregation of fields relevant to ArgoCD health interpretation.
+
+---
+
+
+### 3. Status Aggregation Rules
+
+The aggregation logic depends on the **workload object kind**.
+
+#### Case 1 – Known Workload Kinds
+If the workload kind is **recognized by ArgoCD** (for example Deployment, StatefulSet, DaemonSet, ReplicaSet), the controller applies **predetermined field aggregation rules** consistent with ArgoCD’s native health evaluation.  
+
+Examples:
+- **Deployment:** mark `Available=True` when `availableReplicas == replicas`.
```

Comment:
Statements like this need to explicitly deal with the possibility of different results from different WECs. E.g.,

> Condition "Available" is set to "True" iff `status.availableReplicas` in every WEC equals `spec.replicas` in the WDS; set to "False" iff (you have to write this too!); otherwise set to "Unknown".